lib/iris/cube.py

```diff
@@ -790,6 +800,285 @@ def _is_single_item(testee):
+
+    #
+    # Ensure that the stored local/global dictionaries are "LimitedAttributeDicts".
+    #
+    @staticmethod
+    def _normalise_attrs(
+        attributes: Optional[Mapping],
+    ) -> LimitedAttributeDict:
+        # Convert an input attributes arg into a standard form.
+        # N.B. content is always a LimitedAttributeDict, and a deep copy of input.
+        # Allow arg of None, etc.
+        if not attributes:
+            attributes = {}
+        else:
+            attributes = deepcopy(attributes)
+
+        # Ensure the expected mapping type.
+        attributes = LimitedAttributeDict(attributes)
+        return attributes
+
+    @property
+    def locals(self):
+        return self._locals
+
+    @locals.setter
+    def locals(self, attributes):
+        self._locals = self._normalise_attrs(attributes)
+
+    @property
+    def globals(self):
+        return self._globals
+
+    @globals.setter
+    def globals(self, attributes):
+        self._globals = self._normalise_attrs(attributes)
+
+    #
+    # Provide a serialisation interface
+    #
+    def __getstate__(self):
+        return (self.locals, self.globals)
+
+    def __setstate__(self, state):
+        self.locals, self.globals = state
+
+    #
+    # Support simple comparison, even when contents are arrays.
+    #
+    def __eq__(self, other):
+        # For equality, require both globals + locals to match
+        other = CubeAttrsDict(other)
+        result = self.locals == other.locals and self.globals == other.globals
+        return result
+
+    def __ne__(self, other):
+        return not self == other
+
+    #
+    # Provide a copy method, as for 'dict', but *not* provided by MutableMapping
+    #
+    def copy(self):
+        """
+        Return a copy.
+
+        Implemented with deep copying, consistent with general Iris usage.
+
+        """
+        return CubeAttrsDict(self)
+
+    #
+    # The remaining methods are sufficient to generate a complete standard Mapping
+    # API -- see docs for :class:`collections.abc.MutableMapping`.
+    #
+
+    def __iter__(self):
+        # Ordering: all global keys, then all local ones, but omitting duplicates.
+        # NOTE: this means that in the "summary" view, attributes present in both
+        # locals+globals are listed first, amongst the globals, even though they appear
+        # with the *value* from locals.
+        # Otherwise follows order of insertion, as is normal for dicts.
+        return itertools.chain(
+            self.globals.keys(),
+            (x for x in self.locals.keys() if x not in self.globals),
+        )
+
+    def __len__(self):
+        # Return the number of keys in the 'combined' view.
+        return len(list(iter(self)))
+
+    def __getitem__(self, key):
+        """
+        Fetch an item from the "combined attributes".
+
+        If the name is present in *both* ``self.locals`` and ``self.globals``, then
+        the local value is returned.
+
+        """
```

Comment:
I think this docstring and `__setitem__` are an opportunity to shrink the class and `__init__` docstrings to give this block more structure and make it less intimidating.

I might not make such a fuss but `iris.cube` is one of the most visited API docs pages.

Reply:
As for the previous - See [above](https://github.com/SciTools/iris/pull/5040#discussion_r1269183952)